meta-client/src/main/java/org/apache/helix/metaclient/exception/MetaClientException.java

```diff
@@ -35,4 +35,114 @@ public MetaClientException(String message) {
   public MetaClientException(Throwable cause) {
     super(cause);
   }
+
+  public enum ReturnCode {
+    /** Everything is OK. */
+    OK,
+
+    /** Indicates a system and server-side errors not defined by following codes.
+     * It also indicate a range. Any value larger than this and less than DBUSERERROR
+     * indicating error from server side.*/
+    DB_SYSTEM_ERROR,
+
+    /** Connection to the server has been lost. */
+    CONNECTION_LOSS,
+
+    /** Operation is unimplemented. */
+    UNIMPLEMENTED,
+
+    /** Operation timeout. */
+    OPERATION_TIMEOUT,
+
+    /** Either a runetime or data inconsistency was found. */
+    CONSISTENCY_ERROR,
+
+    /** Session is moved or expired or non-exist. */
+    SESSION_ERROR,
+
+    /** Indicates a system and DB client or a usage errors not defined by following codes.
+     * It also indicate a range. Any value larger than this indicating error from
+     * client or caused by wrong usage.
+     */
+    DB_USER_ERROR,
+
+    /** The listener does not exists."*/
+    INVALID_LISTENER,
+
+    /** Authentication failed.*/
+    AUTH_FAILED,
+
+    /** Invalid arguments */
+    INVALID_ARGUMENTS,
+
+    /** Version conflict.*/
```

Comment:
Curious: what's the difference between BAD_VERSION and CONSISTENCY_ERROR?

Reply:
`CONSISTENCY_ERROR` is thrown by underlying data service (e.g. ZooKeeper) when it finds data inconsistency while handling some user requests, normally during transactional operations.
`BAD_VERSION` is thrown when user call `update` or `set` on a specific version of the entry and the current entry is not on that provided version.